### DISCUSSION TITLE
Non-partition downstream asset runs when partitioned upstream asset is missing some partitions

### DISCUSSION THREAD

**Discussion by javiermas** [dagster-io/dagster#21027] [Q&A]
Imagine I have an upstream dynamically partitioned asset A and a downstream non-partitioned asset B:

```
@asset(
    partitions_def=partitions_def,
)
def asset_a(context):
    repo = S3Repository()
    data = repo.read_parquet(context.partition_key)
    return data

@op
def some_op(asset_a):
    return pd.concat(asset_a)

@graph_asset
def asset_b(asset_a):
    data = some_op(asset_a)
    return data

```

When I'm running locally, I might want to debug asset_b. I'd like to do so by using just one partition of asset_a, which in my case is enough for asset_b to work. However, if I try to materialize asset_b without having materialized all partitions of asset_a I get a FileNotFoundError when dagster tries to read the materializations for all partitions of asset_a. I'd like for dagster to be able to load the specific partition I launched a run for, or at least all available ones, ignoring the ones that aren't available.

I've gone through the documentation, and played a bit with the PartitionMappings to see if I could achieve the desired effect, but I couldn't get it to work.

Does anybody have any ideas?

**Comment by the4thamigo-uk** [comment]
Not sure if `allow_missing_partitions` will work in the case of graph assets...https://github.com/dagster-io/dagster/discussions/14397
